src/main/flight/pid.c

```diff
@@ -335,9 +339,17 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
         }
 
         // -----calculate I component
-        if (motorMixRange < 1.0f) {
-            // Only increase ITerm if motor output is not saturated
-            axisPID_I[axis] += Ki[axis] * errorRate * dT * dynKi * itermAccelerator;
+        const float ITerm = axisPID_I[axis];
+        const float ITermNew = ITerm + Ki[axis] * errorRate * dT * dynKi * itermAccelerator;
+        bool ITermSaturated;
+        if (isTricopter && axis == FD_YAW) {
```

Comment:
To avoid all these checks in pidc. Maybe we should wrap the iterm saturation handling in different functions for different scenarios. Than we can initialise the right function behind a function pointer and even make it more readable